src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1645,6 +1645,7 @@ private static bool IsTypeSpecForTypicalInstantiation(TypeDesc t)
                     //  3) JIT intrinsics - since they have pre-defined behavior
                     devirt = targetMethod.OwningType.IsValueType ||
                         (targetMethod.OwningType.IsDelegate && targetMethod.Name == "Invoke") ||
+                        (targetMethod.OwningType.IsObject && targetMethod.Name == "GetType") ||
```

Comment:
Was there a specific problem you have noticed that made you to add this?

If I am reading the comment in front of this, this check should not be needed since `Object.GetType` is not a virtual method.

Reply:
It was failing on this assert in `Checked` tests: https://github.com/dotnet/runtime/blob/3dc93b37500bf0424178d7d86bdcd36c64003051/src/coreclr/jit/importer.cpp#L8870-L8875
I tried commenting out the assert, but then ran into others so I ended up applying Egor's suggestion: https://github.com/dotnet/runtime/pull/62271#discussion_r761168564, and it passed the checked legs in CI.

Reply:
Ok, sounds good.